tutorials/adding_system_plugins.md

```diff
@@ -0,0 +1,139 @@
+
+```bash
+gz sim ~/gazebo_maritime/models/my_turtle/model.sdf
+```
+
+Hit the play button and you'll see how your turtle falls into the void. Perhaps
+not what you expected but this is completely normal. Gazebo thinks that your
+turtle is in the air without any support underneath. Then, gravity makes your
+turtle to free fall forever.
+
+If we want to simulate that our turtle floats like if it was in the water,
+we'll need to attach a custom buoyancy plugin to our world. This buoyancy plugin
+already exists in Gazebo, we only need to load it.
+
+Now, run Gazebo with the provided `buoyant_turtle.sdf` world and you'll see how
+your turtle does not sink anymore.
+
+```bash
+mkdir -p ~/gazebo_maritime/worlds
+wget https://raw.githubusercontent.com/gazebosim/gz-sim/gz-sim8/tutorials/files/adding_system_plugins/buoyant_turtle.sdf -o ~/gazebo_maritime/worlds/buoyant_turtle.sdf
+export GZ_SIM_RESOURCE_PATH=:$HOME/gazebo_maritime/models
+gz sim -r ~/gazebo_maritime/worlds/buoyant_turtle.sdf
+```
+
+The turtle now stays floating with an oscillating movement up and down.
+
+You just added buoyancy to your model! As a general rule, a maritime model will
+need at least two system plugins:
+
+1. Buoyancy
+2. Hydrodynamics
+
+As you have experienced, the buoyancy plugin generates an upthrust opposing some
+weight of the model. Your model could have positive, neutral or negative
+buoyancy. You'll be able to tune that aspect of your model later.
+
+The hydrodynamics plugin models the force and torque that the vehicle
+experiences when moving within a fluid. Intuitively, the hydrodynamics plugin
+generates drag opposing the movement of the vehicle. If your model does not have
+hydrodynamics, it will behave as if there's no resistance.
+
+For the sake of illustrating the effect of hydrodynamics, let's attach a simple
+controller to our turtle and move it without hydrodynamics. Keep in mind that
+the goal is to move the model one meter and stop it.
+
+Uncomment the following block from `buoyant_turtle.sdf`:
+
+```xml
+<plugin filename="gz-sim-trajectory-follower-system"
+        name="gz::sim::systems::TrajectoryFollower">
+  <link_name>base_link</link_name>
+  <force>0.2</force>
+  <torque>0.01</torque>
+  <range_tolerance>0.1</range_tolerance>
+  <waypoints>
+    <waypoint>1 0</waypoint>
+  </waypoints>
+</plugin>
+```
+
+And run Gazebo:
+
+```bash
+gz sim -r ~/gazebo_maritime/worlds/buoyant_turtle.sdf
+```
+
+As you just observed, we failed in our goal and the turtle behaved as if it was
+moving on ice. Now, let's add hydrodynamics.
+
+Uncomment the following block from `buoyant_turtle.sdf`:
+
+```xml
+<plugin
+  filename="gz-sim-hydrodynamics-system"
+  name="gz::sim::systems::Hydrodynamics">
+  <link_name>base_link</link_name>
+  <xDotU>-0.04876161</xDotU>
+  <yDotV>-1.26324739</yDotV>
+  <zDotW>-1.26324739</zDotW>
+  <kDotP>0</kDotP>
+  <mDotQ>-0.3346</mDotQ>
+  <nDotR>-0.3346</nDotR>
+  <xUabsU>-0.62282</xUabsU>
```

Comment:
Knit: We currently support [added mass in SDFormat](http://sdformat.org/spec?ver=1.10&elem=link#inertial_fluid_added_mass). Perhaps we should add a note that the `hydrodynamics` plugin provides its own added mass fields. If you set these fields you should not set the SDF added mass fields.

On another note we should consolidate the two. Ideally we would deprecate added mass in the hydrodynamics plugin and use the sdf added mass.

Reply:
Thanks, good suggestion! I took the liberty to add your recommendation to the tutorial where we describe the hydrodynamics theory of operation.

https://github.com/gazebosim/gz-sim/pull/2258/commits/2c695048d04fd71a87616de8e8d21490eaafff5b